cdap-docs/developers-manual/source/overview/modes.rst

```diff
@@ -0,0 +1,114 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2014-2015 Cask Data, Inc.
+
+=========================
+CDAP Modes and Components
+=========================
+
+.. _modes-data-application-platform:
+
+Runtime Modes
+=============
+The Cask Data Application Platform (CDAP) can be run in different modes:
+**Distributed CDAP** for staging and production,
+**Standalone CDAP** for testing and development on a developer's laptop,
+or 
+**In-Memory CDAP** for unit testing and continuous integration pipelines.
+
+Regardless of the runtime mode, CDAP is fully-functional and the code you develop never
+changes. However, performance and scale are limited when using in-memory or standalone
+CDAPs. CDAP Applications are developed against the CDAP APIs, making the switch between
+these modes seamless. An application developed using a given mode can easily run in
+another mode.
+
+
+.. _distributed-data-application-platform:
+
+.. rubric:: Distributed CDAP
+
+The **Distributed CDAP** runs in fully distributed mode. In addition to the system components
+of the CDAP, distributed and highly available (HA) deployments of the underlying Hadoop
+infrastructure are included. Production applications should always be run on a Distributed
+CDAP.
+
+See the instructions for either a :ref:`distribution-specific <installation-index>` or 
+:ref:`generic Apache Hadoop <hadoop-index>` cluster for more information.
+
+**Features:**
+
+- A production, staging, and QA mode; runs in a distributed environment
+- Currently uses Apache HBase and HDFS as the underlying storage technology
+- Uses Apache Yarn Containers as the processing abstraction (via Apache Twill)
+
+
+.. _standalone-data-application-platform:
+
+.. rubric:: Standalone CDAP
+
+The **Standalone CDAP** allows you to run the entire CDAP stack in a single Java Virtual
+Machine on your local machine and includes a local version of the CDAP UI. The
+underlying Big Data infrastructure is emulated on top of your local file system. All data
+is persisted.
+
+The Standalone CDAP by default binds to the localhost address, and is not available for
```

Comment:
I'm not sure this is true; a Standalone instance may be available for remote access (from another machine).